packages/eslint-plugin/src/rules/no-floating-promises.ts

```diff
@@ -220,22 +279,33 @@ export default util.createRule<Options, MessageId>({
         // If it is just a property access chain or a `new` call (e.g. `foo.bar` or
         // `new Promise()`), the promise is not handled because it doesn't have the
         // necessary then/catch call at the end of the chain.
-        return true;
+        return { isUnhandled: true };
       } else if (node.type === AST_NODE_TYPES.LogicalExpression) {
-        return (
-          isUnhandledPromise(checker, node.left) ||
-          isUnhandledPromise(checker, node.right)
-        );
+        const leftResult = isUnhandledPromise(checker, node.left);
+        if (leftResult.isUnhandled) {
+          return leftResult;
```

Comment:
[Testing] Oop sorry just noticed - codecov is reporting that this isn't covered by tests. We'll want everything to be tested. Either:

* this is code that can be hit, and a test should be added
* this is code that can't be hit, and should be removed